Chap_Terms.tex

```diff
@@ -4,39 +4,30 @@
+\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or cooperatively at any given time during their execution.
+\item \declareterm{job}\emph{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a \emph{session}. For example, ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' is considered a single \ac{MPMD} job containing two applications.
 \item \declareterm{namespace}\emph{namespace} refers to a character string value assigned by the \ac{RM} to a \textit{job}. All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM}.
-\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or in parallel at any given time during their execution.
-\item \declareterm{rank}\emph{rank} refers to the numerical location (starting from zero) of a process within the defined scope. Thus, {global rank} is the rank of a process within its \emph{job}, while \emph{application rank} is the rank of that process within its \emph{application}.
-\item \declareterm{workflow}\declareterm{workflows}\emph{workflow} refers to an orchestrated execution plan frequently spanning multiple \emph{jobs} carried out under the control of a \emph{workflow manager} process. An example workflow might first execute a computational job to generate the flow of liquid through a complex cavity, followed by a visualization job that takes the output of the first job as its input to produce an image output.
+\item \declareterm{rank}\emph{rank} refers to the ordinal number (starting from zero) assigned to a process within a set of peer processes.  For example, the set of processes could be the members of a job, a compute node, or an application execution. \emph{Global rank} is the rank of a process within its \emph{job}, while \emph{application rank} is the rank of that process within its \emph{application}.
+\item \declareterm{workflow}\declareterm{workflows}\emph{workflow} refers to an orchestrated execution plan frequently spanning multiple \emph{jobs} or \emph{sessions} carried out under the control of a \emph{workflow manager}. An example workflow might first execute a computational job to generate the flow of liquid through a complex cavity, followed by a visualization job that takes the output of the first job as its input to produce an image output.
```

Comment:
spanning multiple sessions -> I don't think it is frequent that workflows are run by multiple users, is it? (as sessions refer to all resources allocated by one user)

Reply:
If the WLM makes a session equal to an allocation then the workflow could span multiple allocations and sessions all tied to the same user.
Maybe that needs to be clarified here.

Reply:
This would then contradict the definition from above, which implies that different sessions belong to different users (see discussion there). Is session and job identical? If not, what is the difference?

Reply:
Which definition implies that different sessions belong to different users?   Session and Job are not identical.   A session refers to a set of resources assigned to a user.  (not necessarily all the resources a user has at one time, but simply one set of resources assigned to a user).  Those resources can be used to create one or more jobs.  Each job is assigned a namespace to name that job.   (In SLURM terminology I think this would map PMIx Session = slurm allocation, PMIx Job = job step)